src/library/scala/concurrent/package.scala

```diff
@@ -170,7 +173,7 @@ package concurrent {
     @throws(classOf[TimeoutException])
     @throws(classOf[InterruptedException])
     final def ready[T](awaitable: Awaitable[T], atMost: Duration): awaitable.type = awaitable match {
-      case f: Future[T] if f.isCompleted => awaitable.ready(atMost)(AwaitPermission)
+      case f: Future[T] if f.isCompleted => awaitable
```

Comment:
It seems we still want to validate the duration

https://github.com/scala/scala/pull/7767 was closed for that reason, in favor of https://github.com/scala/scala/pull/7768

Reply:
Thanks, just back from hometown , will check this weekend

Reply:
I can take a look at this , this weekend.